src/Symfony/Component/HttpFoundation/UriSigner.php

```diff
@@ -37,6 +37,8 @@ public function __construct(#[\SensitiveParameter] string $secret, string $param
      *
      * The given URI is signed by adding the query string parameter
      * which value depends on the URI and the secret.
+     *
+     * @phpstan-pure
```

Comment:
As we use psalm in our CI (even though we use it in an incremental mode), we generally use the `psalm-` prefix rather than the `phpstan-` one.